### PR TITLE
Config model to force-run mgmt commands

### PR DESCRIPTION
**Pull request by iloveagent57**
* Adds a configuration model to help force-run mgmt commands in stage/prod environments, so that
we can work around our task-locking logic in emergencies.
* Installs django-config-models and resolve version incompatibilities for some recently-released major versions.
* No longer runs Django 3 in tox validation, because django-config-models requires Django<3.

https://openedx.atlassian.net/browse/ENT-4206

Testing:
* Run `./manage.py migrate`
* Run any of the management commands e.g. `./manage.py update_content_metadata`
* Run it again to see the task lock.
* Go to http://localhost:18160/admin/catalog/catalogupdatecommandconfig/ and create a new config record, with `enabled` and `force` set to True.  Save it.
* Run the command again, see that it didn't task lock this time.
* Overwrite your config in http://localhost:18160/admin/catalog/catalogupdatecommandconfig/ with a new record to set `force` back to false.

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/models.py [hunk 1]
```diff
@@ -664,3 +665,31 @@ def update_contentmetadata_from_discovery(catalog_query):
...
+        ),
+    )
+
+    @classmethod
+    def current_options(cls):
+        """
+        Returns a dictionary of options from this config model.
+        """
+        current_config = cls.current()
+        if current_config and current_config.enabled:
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/catalog/models.py]: nit: it looks like the `ConfigurationModel` has an `is_enabled` class method on it. Is that something we should use here vs. checking `enabled` ourselves?

https://github.com/edx/django-config-models/blob/master/config_models/models.py#L147

**Review by adamstankiewicz — APPROVED** [review]
LGTM with a non-blocking nit and after testing it locally as well.

File: enterprise_catalog/apps/catalog/models.py (hunk 1 above)
Comment (iloveagent57) [reply enterprise_catalog/apps/catalog/models.py]: Oh good find!  Yeah, let's use that one.

Comment (iloveagent57) [reply enterprise_catalog/apps/catalog/models.py]: Actually, I'm going to leave it as-is: we have to fetch the current instance of the config anyway (i.e. `current_config.force`) and the `enabled` field isn't hidden or anything.  So I could re-write it like this
```python
if cls.is_enabled():
    return {'force': cls.current().force}
return {}
```
but I don't really see an advantage in doing so.  I guess I can reduce `if current_config and current_config.enabled` to just `if current_config.enabled`.

**Pull request merged**